src/ggml.c

```diff
@@ -20220,8 +20205,8 @@ static enum ggml_opt_result ggml_opt_lbfgs(
         ggml_vec_cpy_f32(nx, gp, g);
 
         ls = linesearch_backtracking(&params, nx, x, &fx, g, d, step, xp, f, gb, &cplan, np, ps, &cancel, callback, callback_data);
```

Comment:
Here instead of passing `&cancel`, we should check the return code if it matches `GGML_OPT_CANCEL`